### DISCUSSION TITLE
No matches found for window class when trying to move a window to scratchpad

### DISCUSSION THREAD

**Discussion by TheRedOnline** [i3/i3#5567] [Q&A] [answered]
Attempting to start and move Bitwarden to scratchpad when i3 starts.  Currently in my i3 conf
`for_window [class="Bitwarden"] floating enable`
`for_window [class="Bitwarden"] resize 800 600`
`for_window [class="Bitwarden"] move scratchpad`
`for_window [class="Bitwarden"] border pixel 3`
`exec --no-startup-id bitwarden-desktop`

running xprop on bitwarden
`WM_CLASS(STRING) = "bitwarden", "Bitwarden"`

Starting i3 from fresh session results in i3 opening but not being moved. Running:
`i3-msg for_window [class="Bitwarden"] move scratchpad`
results in 
`zsh: no matches found: [class=Bitwarden]`

It seems that i3 is failing to find the class number.  I have also tried this with instance as well, using the correct instance name.

**Comment by ntcarlson** [comment, marked as answer]
I think you are matching the window correctly, but I see a few other issues:

1. The [syntax for `resize` ](https://i3wm.org/docs/userguide.html#resizingconfig) is wrong (missing `set`).
2. You should probably combine all these rules into a single rule with [command chaining](https://i3wm.org/docs/userguide.html#command_chaining) to ensure they are executed in the right order. E.g.

        for_window [class="Bitwarden"] move scratchpad, border pixel 3, resize set 800 600



Also, you have a few unrelated issues with the `i3-msg` command invoked from the shell:

1. Zsh is interpretting the square brackets in your `i3-msg` command as special characters. That's what is throwing the error you are seeing, not i3. You should quote the entire command to avoid this. You will need to either escape or remove the quotes inside the criteria.
2. `for_window` is a config directive and cannot be called with `i3-msg`. I think what you intended is

        i3-msg "[class=Bitwarden] move scratchpad"

**Reply by TheRedOnline** [reply]
The "unrelated" issue was actually the cause. I had watch some old Luke Smith videos on setting it up, but he wasnt using sxhkd. This is what ultimately fixed it. You were spot on the others too, I had found and fixed most of them by the time I saw this port though. The missing key was the quotes
